Improvements to the announcements cycle

Veldrovive opened the pull request:
I just added a little bit more functionality to make it more user friendly.

Veldrovive committed:
Added functionality where announcements stop cycling on mouse over. Added buttons to manually navigate announcements.

Veldrovive committed:
Changed order to be more logical and easier to understand

HeroCC commented:
Neat! I'm not at a computer now, can you upload some screenshots?

Veldrovive committed:
Fix error that happened if blockSchedule was undefined

Veldrovive commented:
<img width="996" alt="screen shot 2017-08-15 at 2 26 25 pm" src="https://user-images.githubusercontent.com/17326552/29330005-c5746684-81c5-11e7-9b79-6ed58c6425fe.png">
<img width="995" alt="screen shot 2017-08-15 at 2 23 09 pm" src="https://user-images.githubusercontent.com/17326552/29330004-c573b57c-81c5-11e7-86b0-e2e939f5f26a.png">

This is what they look like now, but the one with the pause doesn't fit on small screens. To pause right now you just hover anywhere over the container div. Another option is just to do it as pause on tap the entire div.

HeroCC commented on index.html:
```diff
@@ -23,6 +23,7 @@
   <script src="https://code.jquery.com/jquery-3.2.1.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
   <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
   <script src="https://use.fontawesome.com/a26e812ff2.js"></script>
+  <link href="//netdna.bootstrapcdn.com/font-awesome/3.2.1/css/font-awesome.css" rel="stylesheet">
```
Why do you need this? On the line above it imports fontawesome with some fancy metrics and CDN magic I don't understand or need.

Veldrovive replied:
To be honest, I don't really know why it needs both, but it seems like one covers some symbols and the other covers the rest.

HeroCC replied:
Any idea of what symbols? I am running 4.7, it looks like you're trying to import version 3.2?

Veldrovive replied:
The original doesn't cover the arrows. I'll check 4.7 for arrows that would work as a substitute.

Veldrovive replied:
Ok, it seems that they updated the name of the arrows so I fixed them and removed the import.

HeroCC commented:
Cool, does it work well on mobile? Can you make the buttons a shade lighter & perhaps a bit smaller? And they are hidden when the url is 'https://aspen.studenttech.tk/?billboard'? The billboards all have the '?billboard' at the end, and you can't click on a non interactive screen. And make sure the buttons / title are all the same size, the title area looks much smaller than the buttons.

Also, it would be neat if you had a little circle next to the announcement number that either filled or emptied (going a shade darker) as a timer for how much longer it will be displayed. Something like [this](http://git.wimbarelds.nl/TimeCircles/index.html) but without the text and much smaller? You don't need to use that code (in fact I'd prefer something more lightweight / inhouse).

HeroCC commented:
Also 10+ codacy triggers is far too many plz fix tanks

Veldrovive committed:
Fixed issues codacy had

Veldrovive commented:
I fixed the obvious issues that codacy had, but I didn't un-nest the functions. If you disagree with having nested functions in code, I'll gladly remove them, but they help me with understanding and debugging so I generally use them.

Veldrovive committed:
Accidentally created a new problem while resolving old ones

Veldrovive committed:
Added pause button

Veldrovive committed:
Created system to remove header if billboard parameter is set to true.

Veldrovive committed:
Made announcement buttons work on small screens

Veldrovive committed:
Added a progress bar the announcements.
I made it really thin and red, but I know that some people like it different. If you have a different styling preference, tell me.

Veldrovive committed:
Fixed event [Object object] glitch
This isn't part of the announcement changes, but it needed to be fixed anyways.

Veldrovive committed:
Changed general get query string into a specific check for billboard
Added an event listener for the progress bar

Veldrovive committed:
To appease the codacy gods

Veldrovive committed:
Changed arrows to work with newer font awesome version.
Removed unnecessary import

Veldrovive committed:
Added config variable that describes how long an announcement is on screen.

Veldrovive committed:
Merge remote-tracking branch 'origin/master'

Veldrovive committed:
Added autoReload function

Veldrovive committed:
Merge branch 'master' into master
